internal/schemautil/schemautil.go

```diff
@@ -50,6 +51,36 @@ func OptionalIntPointer(d *schema.ResourceData, key string) *int {
 	return &intValue
 }
 
+// OptionalFloatPointer retrieves a float pointer to a field, if the field is not set, returns nil.
+func OptionalFloatPointer(d *schema.ResourceData, key string) *float64 {
+	val, ok := d.GetOk(key)
+	if !ok {
+		return nil
+	}
+
+	floatValue, ok := val.(float64)
+	if !ok {
+		return nil
+	}
+
+	return &floatValue
+}
+
+// OptionalBoolPointer retrieves a bool pointer to a field, if the field is not set, returns nil.
+func OptionalBoolPointer(d *schema.ResourceData, key string) *bool {
```

Comment:
We already have `ParseOptionalStringToBool` and `ParseOptionalStringToFloat64` can we reuse them?